src/util/async/Operation.hpp

```diff
@@ -149,7 +161,8 @@ struct BasicScheduledOperation {
  * @tparam StopSourceType The type of the stop source
  */
 template <typename RetType, typename StopSourceType>
-class StoppableOperation : public impl::BasicOperation<StoppableOutcome<RetType, StopSourceType>> {
+class StoppableOperation : public impl::BasicOperation<StoppableOutcome<RetType, StopSourceType>>,
+                           public util::WasMoved {
```

Comment:
Shouldn't the `BasicOperation` class inherit `WasMoved` instead?

Reply:
I deliberately inherited on the same level where we actually need the `wasMoved()` call.